### ISSUE TITLE
Ruff requires multiple calls for certain import configurations

### ISSUE THREAD

**Issue by leep-frog** [astral-sh/ruff#14928]
I noticed that I need to call ruff multiple times for the following configuration/code:

Configuration:
```
      lint: {
        select: [
          'I001',
          'I002',
        ],
        isort: {
          'required-imports': [
            "from typing import Dict"
          ],
          'lines-after-imports': 2,
          'combine-as-imports': true,
        },
```

Code:

```python
from typing import Any

def func() -> Any:
  pass
```

After first pass of ruff:

```python
from typing import Dict
from typing import Any


def func() -> Any:
    pass
```

After second pass of ruff:

```python
from typing import Any, Dict


def func() -> Any:
    pass
```

Not sure if there is a config flag to add that would allow this to work in one go.  If not, seems like a bug.

**Comment by dylwil3** [comment]
Hmm... I'm not able to reproduce this. Does it looks like I'm doing everything you're doing?

```console
❯ uv run ruff --version
ruff 0.8.2

❯ cat ruff.toml
[lint]
select = ["I001","I002"]

[lint.isort]
required-imports = ["from typing import Dict"]
lines-after-imports = 2
combine-as-imports = true

❯ cat ex.py
from typing import Any

def func() -> Any:
  pass%                                     
❯ uv run ruff check --fix ex.py
Found 3 errors (3 fixed, 0 remaining).

❯ cat ex.py
from typing import Any, Dict


def func() -> Any:
  pass%  
```

**Comment by leep-frog** [comment]
Ah I see. I'm using the [ruff-wasm-nodejs](https://www.npmjs.com/package/@astral-sh/ruff-wasm-nodejs) npm package. 

Any chance that can be updated to do the same?

**Comment by MichaReiser** [comment]
I consider this out of scope for the JS API. It only provides an API to get all diagnostics for the initial source text. It doesn't even provide an official API for applying fixes. Instead, that's left to the caller. 

You have to keep calling `check` until it returns no more diagnostics with fixes. You also have to be careful, fixes can overlap and some fixes shouldn't be applied when other fixes are present (e.g. one fix might try to remove an import when another edit adds it or even requires it). 

Overall, the JS api has no support for fixes.
